openvdb/points/IndexFilter.h

```diff
@@ -98,6 +126,36 @@ generateRandomSubset(const unsigned int seed, const IntType n, const IntType m)
 } // namespace index_filter_internal
 
 
+template <bool On>
+class ValueMaskFilter
+{
+public:
+    static bool initialized() { return true; }
+    static index::State state() { return index::PARTIAL; }
+    template <typename LeafT>
+    static index::State state(const LeafT& leaf)
+    {
+        if (leaf.isDense())         return On ? index::ALL : index::NONE;
+        else if (leaf.isEmpty())    return On ? index::NONE : index::ALL;
+        return index::PARTIAL;
```

Comment:
Shouldn't this be returning index::PARTIAL if On and at least one voxel is active and vice-versa? i.e:

if (On && leaf.isDense()) return index::ALL;
else if (On && leaf.onVoxelCount()) return index::PARTIAL;
else if (!On && leaf.isEmpty()) return index::ALL;
else if (!On && leaf.offVoxelCount()) return index::PARTIAL;
return index::PARTIAL;

Reply:
Also if this is the case, we need unit tests that catch this as currently everything passes

Reply:
For a start, if you're evaluating *only* active voxels and you know all voxels are inactive, then the result should be index::NONE. Perhaps you meant the return statement you listed to be 'return index::NONE'? If so, I believe that's actually the same logic written differently.

I'll look at the unit tests and make to get this case covered (I should really have included a test case for this filter in TestIndexFilter).